refactor(🔄): use sync queue callback approach to display success toast

jomifepe opened the pull request:
### Description

Alternative approach to https://github.com/prismicio/slice-machine/pull/1436

### How to QA [^1]

<!-- When relevant, describe how to QA your changes. -->

<!-- Your favorite emoji is welcome to close your PR! -->

<!-- A note for reviewers: -->

[^1]:
	Please use these labels when submitting a review:
	:question: #ask:&ensp;Ask a question.
	:bulb: #idea:&ensp;Suggest an idea.
	:warning: #issue:&ensp;Strongly suggest a change.
	:tada: #nice:&ensp;Share a compliment.

jomifepe committed:
refactor(🔄): use sync queue callback approach to display success toast

dani-mp commented on packages/slice-machine/src/features/sync/AutoSyncProvider.tsx:
```diff
@@ -72,79 +73,85 @@ export const AutoSyncProvider: FC<PropsWithChildren> = (props) => {
+            // If the sync failed, we want to display the error message with
+            // the retry button.
+            reject?.(error);
+            throw error;
```
Two things here:
- above if `isUnauthenticatedError(error)`, we're not resolving/rejecting the promise, what should we do here?
- here, i don't think we want to both reject and throw, it's like bubbling the error twice

jomifepe replied:
Yeah makes sense, fixed!

dani-mp commented on packages/slice-machine/src/features/sync/AutoSyncProvider.tsx:
```diff
@@ -42,14 +42,15 @@ export type AutoSyncStatus =
   loggedIn?: boolean;
   changedCustomTypes?: ChangedCustomType[];
   changedSlices?: ChangedSlice[];
+  callback?: () => void;
```
we don't need this, right?

dani-mp commented on packages/slice-machine/src/features/sync/AutoSyncProvider.tsx:
```diff
@@ -42,14 +42,15 @@ export type AutoSyncStatus =
 type AutoSyncContext = {
   autoSyncStatus: AutoSyncStatus;
-  syncChanges: (args?: SyncChangesArgs) => void;
+  syncChanges: (args?: SyncChangesArgs) => void | Promise<void>;
```
why not?
```suggestion
  syncChanges: (args?: SyncChangesArgs) => Promise<void>;
```

jomifepe replied:
Forgot to remove it 😅

jomifepe committed:
feat: use a better label for the toast

jomifepe committed:
fix: display slice add toast only after sync

jomifepe committed:
feat(🍞): display creation toast when adding a new field to a group

fix(tests): skip failing auto-sync tests due to overriding toast

fix: move the save callback out of field sync hook

After discussing with @xrutayisire, the tests could be skipped because
to unblock this feature because we confirmed it was working and the
feature in question is not currently active in production.
